controllers/componentdetectionquery_controller_conditions.go

```diff
@@ -61,9 +69,65 @@ func (r *ComponentDetectionQueryReconciler) SetCompleteConditionAndUpdateCR(ctx
 			Message: fmt.Sprintf("ComponentDetectionQuery failed: %v", completeError),
 		})
 	}
+	updateComponentName(ctx, componentDetectionQuery, r.Client)
```

Comment:
Why not call this in `controllers/update.go` func `updateComponentStub()`?

1. At this point we have two name generation, we do it once in `updateComponentStub()` and then again in `SetCompleteConditionAndUpdateCR()`
2. Seems like we want to keep conditions file simple; its for only setting the status conditions

Reply:
```componentName := "component"
		if len(devfileMetadata.Name) > 0 {
			// use regex to see if metadata.name can be used as component name, since this will be used in kube resources
			if matched := util.CheckWithRegex("^[a-z]([-a-z0-9]*[a-z0-9])?", devfileMetadata.Name); matched {
				componentName = strings.ToLower(devfileMetadata.Name)
			}
		}
		componentName = checkComponentName(componentName, componentDetectionQuery.Status.ComponentDetected)
```

and much further down

```
componentName := "component"
		componentName = checkComponentName(componentName, componentDetectionQuery.Status.ComponentDetected)
```

Reply:
updated. removed the old `checkComponentName`, and use the `getComponentName`